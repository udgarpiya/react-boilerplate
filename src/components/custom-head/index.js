import React from 'react'
import PropTypes from 'prop-types'
import { Helmet } from 'react-helmet'

const CustomHead = ({
  title = 'React Boilerplace',
  description = 'React Boilerplate Description',
}) => {
  return (
    <Helmet>
      <title>{title}</title>
      <meta name="description" content={description} />
    </Helmet>
  )
}

CustomHead.defaultProps = {
  title: 'React Boilerplace',
  description: 'React Boilerplate Description',
}

CustomHead.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
}

export { CustomHead }
export default CustomHead
