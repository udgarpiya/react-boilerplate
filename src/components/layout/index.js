import React from 'react'
import PropTypes from 'prop-types'

import Header from './header'
import Footer from './footer'
import CustomHead from '../custom-head'
import './styles.scss'

const Layout = ({ children, title, description }) => {
  return (
    <div className="layout-component__main-component">
      <CustomHead title={title} description={description} />
      <Header />
      <div className="layout-component__children-container">{children}</div>
      <Footer />
    </div>
  )
}

Layout.defaultProps = {
  title: '',
  description: '',
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
}

export { Layout }
export default Layout
