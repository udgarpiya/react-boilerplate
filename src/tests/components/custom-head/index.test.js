import React from 'react'
import Helmet from 'react-helmet'
import { shallow, mount } from 'enzyme'
import { CustomHead } from 'components/custom-head'

const props = {
  title: 'title',
  description: 'description',
}

describe('Custom Head Component', () => {
  it('It should match snapshot', () => {
    const component = shallow(<CustomHead {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('It should mhave title as "Space X" and description as "description" ', () => {
    const component = mount(<CustomHead {...props} />)

    const helmet = Helmet.peek()
    expect(helmet.title).toBe(props.title)
    expect(helmet.metaTags[0].content).toBe(props.description)
  })
})
