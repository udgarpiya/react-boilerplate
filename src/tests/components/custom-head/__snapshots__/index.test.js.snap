// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Head Component It should match snapshot 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <title>
    title
  </title>
  <meta
    content="description"
    name="description"
  />
</HelmetWrapper>
`;
